### PR TITLE
Product listing

### PR DESCRIPTION
**Pull request by darshandamre**
user can filter products by category, rating and sort by price.

when clicked on a category in home-page, user is navigated to the product-listing page with selected category.
Please take a look at filters, and also check if the same filters are applied when a particular url is shared.

note: cart and wishlist management don't work as of now.

[Preview](https://52shop-web-git-product-listing-darshandamre.vercel.app)

### CODE REVIEW

**Review by ravikumar0403 — APPROVED** [review]
Clean and readable code
Liked the use of react-query and implementation of url search params
good work

File: src/components/Filters/Filters.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+  } = useFilters();
+  const rangeRef = useRef();
+
+  return (
+    <aside className="filters px-4 pt-4">
+      <div className="filter__header">
+        <h3 className="text-md m-0">Filter</h3>
+        <button
+          onClick={() => {
+            rangeRef.current.value = 10000;
```
Comment (tanishk15sharma) [inline src/components/Filters/Filters.js:19]: Here , I liked the way you are using `useRef` .

File: src/components/Filters/Filters.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+              </label>
+            </li>
+          ))}
+        </ul>
+      </section>
+
+      <section className="my-3">
+        <h4 className="text-md mb-1">Rating</h4>
+        <ul>
+          {[4, 3, 2, 1].map(rating => (
```
Comment (tanishk15sharma) [inline src/components/Filters/Filters.js:120]: Even using ratings and catagory list `dynamically` by using `.map` looks exciting to me .

**Review by tanishk15sharma — APPROVED** [review]
The code is written while keeping all the good practices in mind , I Liked how you are using `useRef` , `useLocation` and even Destructuring `Switch` function parameters .

**Pull request merged**